File: src/test/java/kitchenpos/order/eatinorder/infra/persistence/FakeOrderTableRepository.java
package kitchenpos.order.eatinorder.infra.persistence;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;
import java.util.Optional;
import java.util.UUID;
import kitchenpos.order.eatinorder.domain.model.OrderTable;
import kitchenpos.order.eatinorder.domain.repository.OrderTableRepository;

public class FakeOrderTableRepository implements OrderTableRepository {

    private final Map<UUID, OrderTable> storage;

    public FakeOrderTableRepository(Map<UUID, OrderTable> storage) {
        this.storage = storage;
    }

    @Override
    public OrderTable save(OrderTable orderTable) {
        UUID id = UUID.randomUUID();
        orderTable.setId(id);
        storage.put(id, orderTable);
        return orderTable;
    }

    @Override
    public Optional<OrderTable> findById(UUID id) {
        return Optional.ofNullable(storage.get(id));
    }

    @Override
    public List<OrderTable> findAll() {
        return new ArrayList<>(storage.values());
    }
}

File: src/test/kotlin/kitchenpos/order/tobe/eatinorder/infra/FakeOrderTableRepository.kt
package kitchenpos.order.tobe.eatinorder.infra

import java.util.*
import kitchenpos.order.tobe.eatinorder.domain.ordertable.OrderTable
import kitchenpos.order.tobe.eatinorder.domain.ordertable.OrderTableRepository


class FakeOrderTableRepository(
    private val orderTables: MutableMap<UUID, OrderTable> = mutableMapOf()
) : OrderTableRepository {
    override fun save(orderTable: OrderTable): OrderTable {
        orderTables[orderTable.id] = orderTable
        return orderTable
    }

    override fun findById(id: UUID): Optional<OrderTable> {
        return Optional.ofNullable(orderTables[id])
    }

    override fun findAll(): List<OrderTable> {
        return orderTables.values.toList()
    }
}
